### PR TITLE
add workflow to update hacl-packages

### PR DESCRIPTION
**Pull request by pnmadelaine**
When a PR is opened (or updated), this workflow will create (or update) a branch on cryspen/hacl-packages with the corresponding code update.
~~The workflow will then post a link to the branch on the hacl-star PR.~~
If the PR is from branch `[ref]`, then the branch on hacl-packages will be called `hacl-star-[ref]`.
A second workflow deletes the branch when the PR is closed.

### CODE REVIEW

File: .github/workflows/update-hacl-packages.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          git config --local user.name "Hacl Bot"
+          git config --local user.email "hacl-star@mailo.com"
+          git checkout -b $branch
+          git add src include
+          git commit -m "[CI] update code"
+          git push --force --set-upstream  origin $branch
+      - name: comment pr
+        uses: thollander/actions-comment-pull-request@v1
+        with:
+          message: "[CI] corresponding branch on hacl-packages: https://github.com/cryspen/hacl-packages/tree/hacl-star-${{ github.head_ref }}"
```
Comment (franziskuskiefer) [inline .github/workflows/update-hacl-packages.yml:42]: Let's link to the action triggered on this branch. That's what people should look at.
I'd be fine doing this in a followup. Nothing is triggered yet anyway.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks for getting this started.

Let's add a job that's triggered when a PR is closed or merged to delete the branch. Shouldn't be too difficult and I'd like to avoid all the dangling branches in hacl-packages.

Otherwise I'm fine with this. The exact link to the GH action can be added later.

File: .github/workflows/update-hacl-packages.yml (hunk 1 above)
Comment (pnmadelaine) [reply .github/workflows/update-hacl-packages.yml]: I removed it for now

**Comment by msprotz** [comment]
Looks good!

**Review by R1kM — APPROVED** [review]
Looks good to me, thanks.
Do you already have the tokens needed on the hacl-packages side?

File: .github/workflows/hacl-packages-create-branch.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
-name: Update hacl-packages
+name: Create branch on hacl-packages
 
 on:
   pull_request:
     branches: [ main ]
```
Comment (franziskuskiefer) [inline .github/workflows/hacl-packages-create-branch.yml:5]: Should this get a `reopened` as well to create the branch again?

**Review by franziskuskiefer — COMMENTED** [review]
Thanks for adding the delete flow.

File: .github/workflows/hacl-packages-create-branch.yml (hunk 2 above)
Comment (pnmadelaine) [reply .github/workflows/hacl-packages-create-branch.yml:5]: I just tested on a fork and, indeed, reopening a PR does not trigger the worflow.

**Review by franziskuskiefer — APPROVED** [review]
thanks, let's get it in and do the hacl-packages workflow and link it here in the comment.

**Pull request merged**